third_party/2and3/tornado/web.pyi

```diff
@@ -1,11 +1,19 @@
-from typing import Any
+import sys
+
+from typing import Any, Callable, Optional
 from tornado import httputil
 
 MIN_SUPPORTED_SIGNED_VALUE_VERSION: Any
 MAX_SUPPORTED_SIGNED_VALUE_VERSION: Any
 DEFAULT_SIGNED_VALUE_VERSION: Any
 DEFAULT_SIGNED_VALUE_MIN_VERSION: Any
 
+if sys.version_info[:2] >= (3, 5):
+    from typing import Awaitable
+    _MethodType = Callable[..., Optional[Awaitable[None]]]
```

Comment:
Why does the return type have `None` in Python 3.5+ but is `Any` in earlier versions? Is it because we can't represent the type in prior Python versions?

Reply:
I am not 100% sure but IIUC the API logic changed at some point. At least on Python 2 these methods used to return something in subclasses (like `self.render()`), while on Python 3 they act via side effects only (like `self.write()`).